3_javascript/javascript1/week4/script.js

```diff
@@ -0,0 +1,113 @@
+// Voice Assistant
+let savedName = null;
+let todos = [];
+
+function getReply(command) {
+  const Regex = /Hello my name is (.+)/i;
+  const matchName = command.match(Regex);
+  const commandLower = command.toLowerCase();
+  if (matchName) {
+    // Check name already saved------------------
+    const name = matchName[1];
+    if (savedName === name) {
+      return `Welcome back, ${name}!`;
+    } else {
+      savedName = name;
+      return `Nice to meet you, ${name}!`;
+    }
+  } else if (commandLower.includes("what is my name")) {
+    // Check name & ask --------------------------
+    if (savedName) {
+      return `Your name is ${savedName}.`;
+    } else {
+      return "I don't know your name yet.";
+    }
+  } else if (commandLower.startsWith("add ")) {
+    //add todos list-----------------------------
+    const todo = command.slice(4).replace("to my todo", "").trim();
+    todos.push(todo);
+    return `${todo} added to your todo`;
+  } else if (commandLower.startsWith("remove ")) {
+    const todo = command.slice(7).replace("from my todo", "").trim();
+    const index = todos.indexOf(todo);
+    if (index !== -1) {
+      todos.splice(index, 1);
+      return `Removed ${todo} from your todo`;
+    } else {
+      return `${todo} is not in your todo list.`;
+    }
+    //add 4 minutes timer-------------------------
+  } else if (commandLower.includes("set a timer for 4 minutes")) {
+    return setTimer();
+  } else if (commandLower.startsWith("open website ")) {
+    const website = command.slice(13).trim();
+    return `Visit the website: <a href="${website}" target="_blank">${website}</a>`;
+  } else {
+    return "I'm sorry, I didn't understand that command.";
+  }
+}
+
+// (function)------what is today date?---------------------------
+function getCurrentDate() {
+  const months = [
+    "January",
+    "February",
+    "March",
+    "April",
+    "May",
+    "June",
+    "July",
+    "August",
+    "September",
+    "October",
+    "November",
+    "December",
+  ];
+
+  const currentDate = new Date();
+  const day = currentDate.getDate();
+  const month = months[currentDate.getMonth()];
+  const year = currentDate.getFullYear();
+
+  return `${day}. of ${month} ${year}`;
+}
+console.log(getCurrentDate());
+
+//(function)------set timer----------------
+let timer;
+function setTimer() {
+  clearTimeout(timer);
+
+  timer = setTimeout(() => {
+    console.log("Timer done");
+  }, 4 * 60 * 1000);
+
+  return "Timer set for 4 minutes.";
+}
+
+console.log(getReply("Hello my name is Benjamin"));
+console.log(getReply("What is my name?"));
+console.log(getReply("Add fishing to my todo"));
+console.log(setTimer(4));
+console.log(getReply("Open website www.example.com"));
+
+//Simple Arithmetic Calculation-----------  //there is somthing wrong in this code it does't work.
```

Comment:
In your HTML file, the input elements for numbers and the result field are missing the id attribute. Without the id, your JavaScript code won't be able to recognise which elemnt you are trying to get.